src/cases/containers/case-list/case-list.component.spec.ts

```diff
@@ -350,6 +350,49 @@ describe('CaseListComponent', () => {
     });
   });
 
+  describe('Should show share case button', () => {
+    let selectedCases: SearchResultViewItem[] = [];
+    beforeEach(() => {
+      selectedCases = [{
+        case_id: '1',
+        case_fields: {
+          PersonFirstName: 'James',
+          PersonLastName: 'Parker',
+          PersonAddress: '123, Fake Street, Hexton, England, HX08 UTG'
+        }
+      }, {
+        case_id: '2',
+        case_fields: {
+          PersonFirstName: 'Steve',
+          PersonLastName: 'Harris',
+          PersonAddress: 'Davidson House, Forbury Square, Reading, RG1 3EB'
+        }
+      }];
+    });
+    it('Should receive selected cases', () => {
+      component.retrieveSelections(selectedCases);
+      expect(component.selectedCases.length).toEqual(2);
+    });
+    it('Should see the \'Share case\' button greyed out', () => {
+      selectedCases = [];
+      component.retrieveSelections(selectedCases);
+      expect(fixture.debugElement.nativeElement.querySelector('#btn-share-button').textContent).toContain('Share case');
+      expect(component.checkIfButtonDisabled()).toBeTruthy();
+    });
+    it('Share a case button is selectable when any case is selected', () => {
+      component.retrieveSelections(selectedCases);
+      expect(component.checkIfButtonDisabled()).toBeFalsy();
+    });
+    it('Should save share cases to store', () => {
+      component.retrieveSelections(selectedCases);
+      component.shareCaseSubmit();
+      expect(component.checkIfButtonDisabled()).toBeFalsy();
```

Comment:
Can we have validation to check cases saved to store.

Reply:
that has been in:
src/cases/store/selectors/share-case.selectors.spec.ts

Reply:
share-case.selectors.spec.ts is validating store specific functionality.

For this component method shareCaseSubmit(), good to have its own validation. As i see the test, if we comment out code from shareCaseSubmit() method this test would still pass defeating the purpose of this test as we not validating expected output from this method. Correct me if i am wrong.

Reply:
if you comment out the code from shareCaseSubmit() method in ts, the spec.ts will fail to compile, that is what the coverage should be.
if you comment out the  shareCaseSubmit() from html, yes correct the test will still be passing, but I'm not sure whether we need to have such a detailed and high rate coverage test on html as well. Hope that make sense.

Reply:
Not sure if the test that is passing is valid if method shareCaseSubmit() does nothing. See screenshot. 
There is lots of focus on unit tests coverage. please see unit tests section 
 https://tools.hmcts.net/confluence/display/RQA/QA+Review+-+ExUI . "Scope of tests will include valid, invalid & edge cases "

![image](https://user-images.githubusercontent.com/52752138/84494797-03674380-aca2-11ea-9fce-d8767ae6a351.png)